platform/mikro-e/dev/motion-click.c

```diff
@@ -0,0 +1,119 @@
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * \file
+ * 	Motion click driver.
+ *	http://www.mikroe.com/click/motion/
+ *
+ */
+
+#include <lib/sensors.h>
+#include <pic32_gpio.h>
+#include "motion-click.h"
+
+static struct timer debouncetimer;
+static int motion_sensor_status;
+static int motion_sensor_value;
+/*---------------------------------------------------------------------------*/
+static void motion_sensor_read(const struct sensors_sensor *s)
+{
+  if (!motion_sensor_value) {
+    if (timer_expired(&debouncetimer)) {
+    	motion_sensor_value = 1;
+      /* Set a timer for 100ms to ignore false notifications */
+       timer_set(&debouncetimer, CLOCK_SECOND / 10);
+      /* Notify processes that Motion has been detected */
+      sensors_changed(s);
+    }
+  } else {
+    if (timer_expired(&debouncetimer)) {
+    	motion_sensor_value = 0;
+      /* Set a timer for 100ms to ignore false notifications */
+      timer_set(&debouncetimer, CLOCK_SECOND / 10);
+    }
+  }
+}
+
+void motion_sensor_isr(void)
+{
+  motion_sensor_read(&motion_sensor);
+  MOTION_SENSOR_CLEAR_IRQ();
+}
+/*---------------------------------------------------------------------------*/
+static int
+motion_sensor_configure(int type, int value)
+{
+  switch(type) {
+    case SENSORS_HW_INIT:
+      /* Configure gpio pins and initialize interrupt */
+      GPIO_CONFIGURE_AS_INPUT(MOTION_SENSOR_PORT, MOTION_SENSOR_PIN);
+      MOTION_SENSOR_IRQ_INIT();
+      motion_sensor_value = 0;
+      return 1;
+
+    case SENSORS_ACTIVE:
+      if(value) {
+        if(!motion_sensor_status) {
+          /* Enable interrupt for button press */
+          timer_set(&debouncetimer, 0);
+          MOTION_SENSOR_IRQ_ENABLE();
+          motion_sensor_status = 1;
+        }
+      } else {
+        /* Disable interrupt for button press */
+        MOTION_SENSOR_IRQ_DISABLE();
+        motion_sensor_status = 0;
+      }
+      return 1;
+
+    default:
+      return 0;
+   }
+}
+/*---------------------------------------------------------------------------*/
+static int
+motion_sensor_status1(int type)
+{
+  return motion_sensor_status;
+}
+/*---------------------------------------------------------------------------*/
+static int
+motion_sensor_value1(int type)
```

Comment:
use motion_sensor_value here, can change the name of the variable with same name to motion_sensor_state